graylog-storage-elasticsearch7/src/test/java/org/graylog/storage/elasticsearch7/views/ElasticsearchBackendTest.java

```diff
@@ -93,4 +110,48 @@ public void executesSearchForEmptySearchTypes() throws Exception {
         assertThat(queryResult.executionStats()).isNotNull();
         assertThat(queryResult.errors()).isEmpty();
     }
+
+    @Test
+    public void generatedContextHasQueryThatIncludesSearchFilters() {
+        final ImmutableList<UsedSearchFilter> usedSearchFilters = ImmutableList.of(
+                InlineQueryStringSearchFilter.create("", "", "method:GET"),
+                ReferencedQueryStringSearchFilter.create("12345")
+        );
+        doReturn(ImmutableSet.of("method:GET", "method:POST")).when(usedSearchFiltersToQueryStringsMapper).map(usedSearchFilters);
+
+        final Query query = Query.builder()
+                .id("queryWithSearchFilters")
+                .query(ElasticsearchQueryString.of(""))
+                .filters(usedSearchFilters)
+                .timerange(RelativeRange.create(300))
+                .build();
+        final Search search = Search.builder().queries(ImmutableSet.of(query)).build();
+        final SearchJob job = new SearchJob("deadbeef", search, "admin");
+
+        final ESGeneratedQueryContext queryContext = backend.generate(job, query, new SearchConfig(Period.ZERO));
+        final QueryBuilder esQuery = queryContext.searchSourceBuilder(new SearchType.Fallback()).query();
+        assertThat(esQuery)
+                .isNotNull()
+                .isInstanceOf(BoolQueryBuilder.class);
+
+        final List<QueryBuilder> filters = ((BoolQueryBuilder) esQuery).filter();
+
+        //filter for empty ES query
+        assertThat(filters.stream()
+                .anyMatch(queryBuilder -> queryBuilder instanceof MatchAllQueryBuilder))
+                .isTrue();
+
+        //2 filters from search filters
+        assertThat(filters.stream()
```

Comment:
We have discussed some possible simplifications of these assertions. Could you please implement them?